todo_app/tests_e2e/test_e2e.py

```diff
@@ -0,0 +1,85 @@
+import pytest
+from selenium import webdriver
+from threading import Thread
+from todo_app import app
+from dotenv import load_dotenv, find_dotenv   
+import time
+import os
+from todo_app import app
+import requests
+
+@pytest.fixture(scope='module')
+def test_app():
+       file_path = find_dotenv('.env')
+       load_dotenv(file_path, override=True)
+       board_id = create_trello_board()
+       os.environ['ITEMS_BOARD_ID'] = board_id
+       os.environ['TODO_LIST_ID'] = get_list_id_for_board(board_id, "TO DO")
+       os.environ['DONE_LIST_ID'] = get_list_id_for_board(board_id, "DONE")
+
+       #Start the application in its own thread
+       application = app.create_app()
+       thread = Thread(target=lambda: application.run(use_reloader=False))
+       thread.daemon = True
+       thread.start()
+       yield application
+
+       #Delete the test board
+       thread.join(1)
+       time.sleep(5)
+       delete_trello_board(board_id)
+
+@pytest.fixture(scope="module")
+def driver():
+    with webdriver.Firefox() as driver:
+        yield driver
+
+def testDriver(test_app, driver):
+    driver.get('http://127.0.0.1:5000/')
+    assert driver.title == 'To-Do App'
+    
+def create_trello_board():
+    API_KEY = os.getenv('TRELLO_API_KEY')
+    TOKEN = os.getenv('TRELLO_API_TOKEN')
+    url = f"https://api.trello.com/1/boards/"
+    query = {"key": API_KEY, "token": TOKEN, "name": 'TestBoard'}
+    response = requests.request("POST", url, params=query)
+    value = response.json()["id"]
+    print("Test board created, id=" + value)
+    return value
+
+def delete_trello_board(id):
+    API_KEY = os.getenv('TRELLO_API_KEY')
+    TOKEN = os.getenv('TRELLO_API_TOKEN')
+    url = "https://api.trello.com/1/boards/" + id
+    query = {"key": API_KEY, "token": TOKEN}
+    requests.request("DELETE", url, params=query)
+    print("Test board deleted, id=" + id)
+
+def test_createTask(test_app, driver):
+    testDriver(test_app, driver)
+    driver.find_element_by_id("title").send_keys("Selenium is flaky")
+    driver.find_element_by_id("description").send_keys("Selenium is flaky")
+    driver.find_element_by_id("submit").click()
+    time.sleep(5)
```

Comment:
Sleeping isn't a major problem here, but can quickly become a significant problem in testing pipelines.

Your e2e test is well structured to take advantage of the natural waiting that selenium can do; in your driver fixture you could update the wait setting with:
```python
    with webdriver.Firefox() as driver:
        driver.implicitly_wait(5)
        yield driver
```

And that should allow you to remove your sleep statements

Reply:
Changes made as suggested. Thanks.